### ISSUE TITLE
Out of bounds Exception in iOS9

### ISSUE THREAD

**Issue by mheicke** [modocache/MDCDamerauLevenshtein#2]
The function mdc_damerauLevenshteinDistance in MDCDamerauLevenshteinDistance.m calls MDCDistanceMatrixWalk with the parameter y = 1. In the given MDCDistanceMatrixWalker mdc_sameCharacterAtLeft:index:right is called with the parameter y-2. Since y is of type NSUInteger and y = 1, y-2 equals 18446744073709551615 on a 64bit system. Since y is used with "characterAtIndex in mdc_sameCharacterAtLeft..." it leds to an out of bounds exception ("[NSTaggedPointerString characterAtIndex:]: Index 18446744073709551615 out of bounds; string length 3"

left = mdc
leftIndex = 18446744073709551615
right = schumacher
rightIndex = 5

**Comment by mheicke** [comment]
with a little more research, it seems like the "mdc_normalizeDistanceParameters" function might be the problem since it does swap the values differently to iOS8, if my observations are correctly

**Comment by mheicke** [comment]
I just experimented a little bit. the problem seems to only occur when your classes are included via cocoapod. For myself I added "x > 1 && y > 1" to

``` objective-c
        if ([NSString mdc_sameCharacterAtLeft:left index:y-1 right:right index:x-2] &&
            [NSString mdc_sameCharacterAtLeft:left index:y-2 right:right index:x-1]) {
            transposition = matrix->distances[y-2][x-2] + cost;
        }
```

hence

``` objective-c
        if (x > 1 && y > 1 && [NSString mdc_sameCharacterAtLeft:left index:y-1 right:right index:x-2] &&
            [NSString mdc_sameCharacterAtLeft:left index:y-2 right:right index:x-1]) {
            transposition = matrix->distances[y-2][x-2] + cost;
        }
```

which should come with little cost. To be honest, I'm not 100% sure if that changes any of the algorithm, but I think it does not.

**Comment by modocache** [comment]
Fixed in #3. Thanks, @mheicke!
